app/frontend/wallet/account-manager.ts

```diff
@@ -46,29 +49,66 @@ const AccountManager = ({
   }
 
   async function discoverAccounts() {
+    if (
+      maxAccountIndex === accounts.length - 1 || accounts.length > 0
+        ? !(await accounts[accounts.length - 1].isAccountUsed())
+        : false
+    ) {
+      return accounts
+    }
+
```

Comment:
I think it can be rewritten like this, which would look more natural to me:
```suggestion
    if (
      (maxAccountIndex === accounts.length - 1)
      || (accounts.length > 0 && !(await accounts[accounts.length - 1].isAccountUsed()))
    ) {
      return accounts
    }
```

Reply:
Refactored.